compiler/src/dotty/tools/dotc/transform/patmat/Space.scala

```diff
@@ -315,6 +315,8 @@ class SpaceEngine(implicit ctx: Context) extends SpaceLogic {
       else
         Typ(ConstantType(c), false)
     case _: BackquotedIdent => Typ(pat.tpe, false)
+    case Ident(nme.WILDCARD) =>
+      Or(Typ(pat.tpe.stripAnnots, false) :: Typ(ConstantType(Constant(null))) :: Nil)
```

Comment:
Would be cleaner to have  `val nullType = ConstantType(Constant(null))` somewhere.